### PR TITLE
[WIP] Pr uavcan v1 node client

### PR DESCRIPTION
**Pull request by Igor-Misic**
Next step for integrating uavcan_v1

Todo:
- add options to start/connect with more than one CAN bus [@Igor-Misic ]

Know issues:
- socketcan driver will hang if the node is not properly connected to the CAN bus (reproducible if CAN wires are not connected or there is no active CAN device on the bus)
- uavcan node client can't be properly reconnected if it is rebooted. 
The problem is on the server-side because canard remembers inside `_rx_subscriptions` different `rxs->transfer_id` from the previous session.

### CODE REVIEW

**Comment by PetervdPerk-NXP** [comment]
Looks like a nice starting point for a C++ based cannode implementation.
At a first glance I've got some small remarks
- I think PX4 uses the hpp extension for C++ headers
- @dagar already made a C++ abstraction for the CAN interface you can check this out, maybe that would be nice for the next step https://github.com/PX4/PX4-Autopilot/blob/master/src/drivers/uavcan_v1/CanardInterface.hpp

File: .gitmodules [hunk 1]
```diff
@@ -60,6 +60,12 @@
 [submodule "src/drivers/uavcan_v1/public_regulated_data_types"]
 	path = src/drivers/uavcan_v1/public_regulated_data_types
 	url = https://github.com/UAVCAN/public_regulated_data_types.git
+[submodule "src/drivers/uavcan_v1_node_client/libcanard"]
+	path = src/drivers/uavcan_v1_node_client/libcanard
+	url = https://github.com/UAVCAN/libcanard.git
+[submodule "src/drivers/uavcan_v1_node_client/public_regulated_data_types"]
```
Comment (dagar) [inline .gitmodules:66]: We should centralize these.

**Comment by Igor-Misic** [comment]
No need for this for now. It will be reopened if needed.

**Comment by Igor-Misic** [comment]
No need for this for now. It will be reopened if needed.

**Pull request closed without merging**